src/foam/nanos/auth/Address.js

```diff
@@ -149,42 +139,31 @@ foam.CLASS({
       name: 'streetName',
       width: 70,
       documentation: 'The structured field for the street name of the postal address.',
-      validateObj: function(streetName) {
-        if ( streetName.trim() === '' ) {
-          return 'Street name required.';
-        }
-        var streetNameRegex = /^[a-zA-Z0-9 ]{1,70}$/;
-        if ( ! streetNameRegex.test(streetName) ) {
-          return 'Invalid street name.';
+      validationPredicates: [
+        {
+          args: ['structured', 'streetName'],
+          predicateFactory: function(e) {
+            return e.OR(
+              e.EQ(foam.nanos.auth.Address.STRUCTURED, false),
+              e.REG_EXP(foam.nanos.auth.Address.STREET_NAME, /^\s*.+\s*$/)
```

Comment:
I would rather have the preSet() do a trim(), or add a trim: true property to String properties and then just use EQ here. REG_EXP is big and slow for this and doesn't clean up the string.

Reply:
Actually, we already have a trim: property on Strings.

Reply:
We don't want active trimming on the value because then you can't add spaces while filling it out in an input.